src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -812,10 +812,10 @@ def get_realm_tokens(self):
     @UpdatePermission
     @allow_empty_body
     # pylint: disable=W0613
-    def import_realm_token(self, realm_token, zone_name, daemon_name=None):
+    def import_realm_token(self, realm_token, zone_name, port, service_spec, daemon_name=None):
         try:
             multisite_instance = RgwMultisite()
-            result = CephService.import_realm_token(realm_token, zone_name)
+            result = CephService.import_realm_token(realm_token, zone_name, port, json.loads(service_spec))
```

Comment:
why not send this as a json parsed string rather than a json object? That could remove some of the extra type-checking you are doing in the module right?
```suggestion
            result = CephService.import_realm_token(realm_token, zone_name, port, service_spec)
```

Reply:
@nizamial09 , Updated the PR